client/rpmtrans.c

```diff
@@ -678,9 +692,9 @@ TDNFRunTransaction(
     return dwError;
 
 error:
-    if(pTS && dwError != ERROR_TDNF_RPM_CHECK)
+    if(pTS)
     {
-        doCheck(pTS);
+        reportProblems(pTS);
```

Comment:
I wonder if we should move this to right after the two `rpmtsRun()` calls. If it's true that only those two generate problems.

Reply:
I'm not sure I follow. Are you suggesting moving it to just before the `cleanup` label? Or else placing a call to `reportProblems` in each of the individual failure paths near the failed call sites? Is there a reason that we need to avoid calling it in other cases? I do appreciate that, with where it is now, changes to the core logic in the body of the function won't introduce cases where something could fail above and generate problems without the problems being reported on.

With that said, if you have guidance as to what you're looking for or a change that you'd like me to make, please do let me know.

Reply:
> I'm not sure I follow. Are you suggesting moving it to just before the cleanup label?

No

> Or else placing a call to reportProblems in each of the individual failure paths near the failed call sites?

Yes, this.

> Is there a reason that we need to avoid calling it in other cases?

Yes. We get to the error label for all sorts of errors, including memory allocation errors or other system errors. Thus my comment in the code ("Did you make this change to avoid...). But we only want to report problems when there are actually rpm problems.

My comment is outside of the scope of the issue you are addressing though - it's independent, we could address it in another change.

Reply:
Okay, I understand now. Thanks for clarifying. Would you like me to file an issue for separate tracking of this feedback? (I'm not sure what process this project follows for out-of-scope PR feedback like this.)

Reply:
Sure, you can do that.

Reply:
Okay, filed as #364.